Error received after "make" command in version 1.10.1

Hello, as I mentioned in the title, it returns an error after the make command (on RHEL9.6):

/home/ircds/znc-1.10.1/modules/modperl/modperl_biglib.cpp: In function ‘SV* std_set_Sl_CString_Sg__keys_(std::set<CString>*)’:
/home/ircds/znc-1.10.1/modules/modperl/modperl_biglib.cpp:2296:26: error: ‘newAV_alloc_x’ was not declared in this scope
 2296 |                 AV* av = newAV_alloc_x(self->size());
cc1plus: note: unrecognized command-line option ‘-Wno-reserved-user-defined-literal’ may have been intended to silence earlier diagnostics
make[2]: *** [modules/modperl/CMakeFiles/modperl_lib.dir/build.make:89: modules/modperl/CMakeFiles/modperl_lib.dir/modperl_biglib.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3847: modules/modperl/CMakeFiles/modperl_lib.dir/all] Error 2
make: *** [Makefile:136: all] Error 2


My CMake command is also this and I have not received an error before:

cmake3 -DCMAKE_INSTALL_PREFIX="$HOME/.znc" -DWANT_PERL=YES -DWANT_PYTHON=YES -DWANT_TCL=YES -DWANT_IPV6=YES

--- a/modules/modperl/modperl.i
+++ b/modules/modperl/modperl.i
@@ -82,7 +82,9 @@ namespace std {
                 return i != self->end();
             }
             SV* keys_() {
-                AV* av = newAV_alloc_x(self->size());
+                // TODO: switch to newAV_alloc_x, requires perl 5.35.1
+                AV *av = newAV();
+                av_extend(av, self->size());
                 // assume SCString
                 int i = 0;
                 for (const auto& a : *self) {

